test/boost/alternator_expressions_test.cc

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include <boost/test/unit_test.hpp>
+#include <seastar/testing/test_case.hh>
+#include <seastar/core/coroutine.hh>
+#include "alternator/expressions.hh"
+
+// ANTLR3 leaks memory when it tries to recover from missing token.
+// Reproduces #25878
```

Comment:
Thanks for adding a comment to explain the test.
I would add to it a sentence that this leak was reported by ASAN (?) when running this test in debug mode - the test passed but the leak is discovered when the test file exits (right?).